src/cljs/ataru/hakija/application_handlers.cljs

```diff
@@ -495,3 +504,10 @@
        :http {:method  :delete
               :url     (str "/hakemus/api/files/" key)
               :handler [:application/handle-attachment-delete field-descriptor component-id key]}})))
+
+(reg-event-db
+  :application/remove-attachment-error
+  (fn [db [_ field-descriptor component-id attachment-idx]]
+    (-> db
+        (update-in [:application :answers (keyword component-id) :values] autil/remove-nth attachment-idx)
```

Comment:
Normally when we remove an attachment from the UI (`:application/handle-attachment-delete`), we use the attachment key to find which attachment we're removing, not the index. This is because that code is asynchronously executed after an AJAX call, so the index might have changed during the time the AJAX request was on the wire. 

I don't do that here because this code is not behind an AJAX call. That said, it is not guaranteed to be synchronous either (there's a `dispatch` before this code is executed), and I've already used more time explaining why I use the index here than using the key would have taken me. 🤷‍♀️